blog-posts/github-billing-dashboard.md

```diff
@@ -0,0 +1,169 @@
+
+```
+billing.csv
+
+Date,Product,Repository Slug,Quantity,Unit Type,Price Per Unit,Actions Workflow,Notes
+2021-06-14,actions,sample-repository1,21,UBUNTU,$0.008,.github/workflows/integration-test.yml,
+2021-06-14,shared storage,sample-repository2,20.1028,gb,$0.25,,
+2021-06-15,actions,sample-repository3,26,UBUNTU,$0.008,.github/workflows/main.yml,
+2021-06-15,actions,sample-repository2,17,UBUNTU,$0.008,.github/workflows/integration-test.yml,
+```
+
+```tsx
+export interface UsageReportCsvEntry {
+	date: string;
+	product: string;
+	repositorySlug: string;
+	quantity: string;
+	unitType: string;
+	pricePerUnit: string;
+	actionsWorkflow: string;
+	notes: string;
+}
+
+export interface UsageReportEntry extends UsageReportCsvEntry {
+	totalPrice: number;
+}
+
+parse<UsageReportCsvEntry>(file, {
+	header: true,
+	skipEmptyLines: true,
+	transformHeader: (header: string): string => {
+		return camalize(header);
+	},
+	complete: (result) => {
+			const githubBillingEntries: UsageReportEntry[] = result.data.map(
+				(dailyEntry) => {
+					return {
+					...dailyEntry,
+					totalPrice: *parseFloat*(dailyEntry.quantity) * *parseFloat*(dailyEntry.pricePerUnit),
+				};
+			}
+		);
+	},
+});
+```
+
+### Recharts
+
+Probably the most obvious feature of the dashboard is the functionality to display the costs in charts. For this, we used the React library Recharts. This library comes with some handy features that were also important for this project. Besides the possibility to choose between different chart types (e.g. Bar or Line), Recharts also leaves a lot of room for customizing the charts. This includes, for example, various stylings, additional elements (e.g. tooltip, legend, ...), and the possibility to simply add features that are not available.
+
+```tsx
+<BarChart data={currentData} margin={{ bottom: 20 }}>
+  <CartesianGrid vertical={false} stroke={"rgba(255, 255, 255, 0.1)"} />
+  <XAxis
+    dataKey={groupedBy === "daily" ? "day" : "week"}
+    tick={{ fill: "white" }}
+    axisLine={false}
+    tickLine={false}
+    minTickGap={10}
+  />
+  <YAxis
+    domain={[0, maxValueOfYAxis]}
+    unit=" $"
+    tickCount={maxValueOfYAxis + 1}
+    tick={{ fill: "white" }}
+    axisLine={false}
+    tickLine={false}
+  />
+</BarChart>
+```
+
+### date-fns
+
+For grouping costs by days, weeks and months we had to work a lot with dates. The library `date-fns` made this work easier for us because it brings important functions like `startOfMonth()`, `getISOWeek()`, or `lightFormat()`, which allows you to customize the date format easily. For example, we used the `lightFormat()` function to display the start or end date of the data set. Due to the detailed documentation, it was very pleasant to work with `date-fns`.
```

Comment:
Did you even think about moment? Would be some nice context knowledge, why you decided for date-fns (size, functional approach etc?). Absolutely optional.